app/screens/settings/notification_auto_responder/notification_auto_responder.tsx

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {View} from 'react-native';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+import BlockItem from '@components/block_item';
+import FloatingTextInput from '@components/floating_text_input_label';
+import FormattedText from '@components/formatted_text';
+import {General} from '@constants';
+import {useTheme} from '@context/theme';
+import {t} from '@i18n';
+import {changeOpacity, makeStyleSheetFromTheme, getKeyboardAppearanceFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {getNotificationProps} from '@utils/user';
+
+import type UserModel from '@typings/database/models/servers/user';
+
+const headerText = {
+    id: t('notification_settings.auto_responder'),
+    defaultMessage: 'Custom message',
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg,
+        },
+        wrapper: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.06),
+            flex: 1,
+            paddingTop: 35,
+        },
+        input: {
+            color: theme.centerChannelColor,
+            fontSize: 15,
+            height: 150,
+            paddingHorizontal: 15,
+            paddingVertical: 10,
+        },
+        footer: {
+            paddingHorizontal: 15,
+            fontSize: 12,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+            textAlign: 'justify',
+        },
+        area: {
+            paddingHorizontal: 16,
+        },
+        label: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200, 'Regular'),
+        },
+    };
+});
+type NotificationAutoResponderProps = {
+    currentUser: UserModel;
+}
+const NotificationAutoResponder = ({currentUser}: NotificationAutoResponderProps) => {
+    const autoresponderRef = useRef(null);
+    const userNotifyProps = useMemo(() => getNotificationProps(currentUser), [currentUser.notifyProps]);
```

Comment:
Do we want to handle the defaulting here too, so we don't have to do one extra render?
Also, since `userNotifyProps` is only used to initialize the state, there is no need for it to depend on anything.

Reply:
This is just going to be the skeleton.  I will add the logic in the coming PRs.